### PR TITLE
Extract release notes building from JiraSite

### PR DESCRIPTION
**Pull request by artkoshelev**

### CODE REVIEW

**Comment by artkoshelev** [comment]
> Coverage increased (+0.006%)
🎆

**Comment by artkoshelev** [comment]
@oleg-nenashev so, should i create new branch for major release and do all the refactoring work there?

**Comment by olamy** [comment]
@artkoshelev simply do not break backward compatibility :)

**Comment by oleg-nenashev** [comment]
I do not see a strict need for a breaking change here. A lot of refactoring could be done without breaking API. It's up to plugin maintainers, of course

File: src/main/java/hudson/plugins/jira/JiraSession.java [hunk 1]
```diff
@@ -165,10 +171,6 @@ public Version getVersionByName(String projectKey, String name) {
         return null;
     }
 
-    public List<Issue> getIssuesWithFixVersion(String projectKey, String version) throws TimeoutException {
```
Comment (warden) [inline src/main/java/hudson/plugins/jira/JiraSession.java]: `@Deprecated`

**Comment by warden** [comment]
@artkoshelev would it work to leave a `@Deprecated` `JiraSite.getReleaseNotesForFixVersion()` and call inside it `JiraCreateReleaseNotes.getReleaseNotesForFixVersion()`?

**Comment by artkoshelev** [comment]
i'll give it a try

**Comment by warden** [comment]
@artkoshelev what do you think about rebasing this to `feature/4.0` ?

**Comment by artkoshelev** [comment]
@warden sure, go for it

**Comment by artkoshelev** [comment]
ah, need to resolve confilcts

**Pull request closed without merging**